pubnub-chat-impl/src/jsMain/kotlin/ChatJs.kt

```diff
@@ -318,23 +319,33 @@ class ChatJs internal constructor(val chat: ChatInternal, val config: ChatConfig
     }
 
     fun getUnreadMessagesCounts(params: PubNub.GetMembershipsParametersv2?): Promise<Array<GetUnreadMessagesCountsJs>> {
-        return chat.getUnreadMessagesCounts(
+        return fetchUnreadMessagesCounts(params).then {
+                result ->
+            result.counts
+        }
+    }
+
+    fun fetchUnreadMessagesCounts(params: PubNub.GetMembershipsParametersv2?): Promise<FetchUnreadMessagesCountsJs> {
+        return chat.fetchUnreadMessagesCounts(
             params?.limit?.toInt(),
             params?.page?.toKmp(),
             params?.filter,
             extractSortKeys(params?.sort)
         ).then { result ->
-            result.map { unreadMessagesCount ->
-                createJsObject<GetUnreadMessagesCountsJs> {
-                    this.channel = unreadMessagesCount.channel.asJs(this@ChatJs)
-                    this.membership = unreadMessagesCount.membership.asJs(this@ChatJs)
-                    this.page = MetadataPage(unreadMessagesCount.next, unreadMessagesCount.prev)
-                    this.count = unreadMessagesCount.count.toDouble()
-                }
-            }.toTypedArray()
+            createJsObject<FetchUnreadMessagesCountsJs> {
+                this.counts = result.counts.map { it.toJs() }.toTypedArray()
+                this.page = MetadataPage(result.next, result.prev)
+            }
         }.asPromise()
     }
 
+    private fun GetUnreadMessagesCounts.toJs() =
```

Comment:
I am wondering why this method was not present before. Why it is needed now?

Reply:
This is because the newly introduced `UnreadMessagesCounts` class stores an array of `GetUnreadMessagesCounts`.These stored items must be serialized into the type expected by the JS